src/array.ts

```diff
@@ -14,6 +14,56 @@ export const group = <T, Key extends string | number | symbol>(
   }, {} as Record<Key, T[]>)
 }
 
+/**
+ * Creates an array of grouped elements, the first of which contains the
+ * first elements of the given arrays, the second of which contains the
+ * second elements of the given arrays, and so on.
+ *
+ * Ex. const zipped = zip(['a', 'b'], [1, 2], [true, false]) // [['a', 1, true], ['b', 2, false]]
+ */
+export function zip<T1, T2, T3, T4, T5>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[],
+  array4: T4[],
+  array5: T5[]
+): [T1, T2, T3, T4, T5][]
+export function zip<T1, T2, T3, T4>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[],
+  array4: T4[]
+): [T1, T2, T3, T4][]
+export function zip<T1, T2, T3>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[]
+): [T1, T2, T3][]
+export function zip<T1, T2>(array1: T1[], array2: T2[]): [T1, T2][]
+export function zip<T>(...arrays: T[][]): T[][] {
+  if (!arrays || !arrays.length) return []
+  return new Array(Math.max(...arrays.map(({ length }) => length)))
+    .fill([])
+    .map((_, idx) => arrays.map(array => array[idx]))
+}
+
+/**
+ * Creates an object mapping the specified keys to their corresponding values
+ *
+ * Ex. const zipped = zipToObject(['a', 'b'], [1, 2]) // { a: 1, b: 2 }
+ */
+export function zipToObject<K extends string | number | symbol, V>(
+  keys: K[],
+  values: V[]
+): Record<K, V> {
+  if (!keys || !keys.length || !values || !values.length)
+    return {} as Record<K, V>
+  return keys.reduce(
+    (acc, key, idx) => ({ ...acc, [key]: values[idx] }),
+    {} as Record<K, V>
+  )
+}
```

Comment:
Now that I see this I think it has the potential to be a really powerful function. I don't know if we _need_ it to be, but sharing the thought. What do you think?

```suggestion
export function zipToObject<K extends string | number | symbol, V>(
  keys: K[],
  values: V | ((key: K, idx: number) => V) | V[]
): Record<K, V> {
  if (!keys || !keys.length)
    return {} as Record<K, V>
  const getValue = isFunction(values) 
    ? values 
    : isArray(values) 
      ? (k, i) => values[i] 
      : () => values
  return keys.reduce(
    (acc, key, idx) => ({ ...acc, [key]: getValue(key, idx) }),
    {} as Record<K, V>
  )
}
```

If it was this dynamic maybe we'd call it something less specific. I like `zipo` :) 

Still fits your use case but now it's sort of a dynamic object builder

```ts
zipo(['a', 'b'], [1, 2]) // => { a: 1, b: 2 }
zipo(['a', 'b'], 'x') // => { a: 'x', b: 'x' }
zipo(['a', 'b'], (k, i) => `${k}${i}`) // => { a: 'a1', b: 'b2' }
```

Reply:
Love it! Though not sure why we'd need to rename it – after all, it is still just producing an object haha. I still think at first glance `zipToObject` is clearer than `zipo`, though my preference isn't super strong.

Reply:
Also simplified your code a little bit and added a new test for the callback parameter 👍🏻

Reply:
Ugh your right... haha my mind is desperate to find a way to not name it a camel case name but your totally right.